auction-app-backend/src/main/java/com/example/auctionapp/repository/ProductRepository.java

```diff
@@ -54,4 +54,9 @@ List<ProductEntity> findProductEntityByUserEntity_UserIdAndAndStatusAndBidsCount
     @Query("SELECT COALESCE(MIN(p.startPrice), 0), COALESCE(MAX(p.startPrice), 0) " +
             "FROM ProductEntity p WHERE p.status = 'ACTIVE'")
     List<BigDecimal[]> findMinAndMaxPrices();
+
+    @Query(value = "SELECT p FROM ProductEntity p " +
+            "WHERE p.categoryEntity.categoryId = :categoryId " +
+            "ORDER BY RANDOM() LIMIT 3")
+    List<ProductEntity> findRandomProductsByCategoryId(final UUID categoryId);
```

Comment:
Shouldn't this be featured products, so shouldn't we use the endpoint we already have and the methods we already have?